homeassistant/components/huum/__init__.py

```diff
@@ -2,46 +2,29 @@
 
 from __future__ import annotations
 
-import logging
-
-from huum.exceptions import Forbidden, NotAuthenticated
-from huum.huum import Huum
-
 from homeassistant.config_entries import ConfigEntry
-from homeassistant.const import CONF_PASSWORD, CONF_USERNAME
 from homeassistant.core import HomeAssistant
-from homeassistant.exceptions import ConfigEntryNotReady
-from homeassistant.helpers.aiohttp_client import async_get_clientsession
 
-from .const import DOMAIN, PLATFORMS
+from .const import PLATFORMS
+from .coordinator import HuumDataUpdateCoordinator
 
-_LOGGER = logging.getLogger(__name__)
+type HuumConfigEntry = ConfigEntry[HuumDataUpdateCoordinator]
 
 
-async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+async def async_setup_entry(hass: HomeAssistant, entry: HuumConfigEntry) -> bool:
     """Set up Huum from a config entry."""
-    username = entry.data[CONF_USERNAME]
-    password = entry.data[CONF_PASSWORD]
-
-    huum = Huum(username, password, session=async_get_clientsession(hass))
+    coordinator = HuumDataUpdateCoordinator(
+        hass,
+        config_entry=entry,
```

Comment:
```suggestion
       entry,
```
it's just styling, but either make both args or kwargs